container/go/cmd/create_image_config/createImageConfig.go

```diff
@@ -0,0 +1,413 @@
+	flag.Var(&entrypoint, "entrypoint", "Override the Entrypoint of the previous layer.")
+	flag.Var(&layer, "layer", "Layer sha256 hashes that make up this image.")
+	flag.Var(&stampInfoFile, "stampInfoFile", "A list of files from which to read substitutions to make in the provided fields.")
+	flag.Parse()
+	log.Println("Running the Image Config manipulator...")
+
+	if *outputConfig == "" {
+		log.Fatalln("Required option -outputConfig was not specified.")
+	}
+	if *nullEntryPoint == "true" {
+		entrypoint = nil
+	}
+	if *nullCmd == "true" {
+		command = nil
+	}
+
+	// read config file into struct.
+	configPath, err := os.Open(*baseConfig)
+	if err != nil {
+		log.Fatalf("Failed to read the parent image's config file: %v", err)
+	}
+	configFile, err := v1.ParseConfigFile(configPath)
+	if err != nil {
+		log.Fatalf("Failed to successfully parse config file json contents: %v", err)
+	}
+
+	// read manifest file into struct if provided.
+	if *baseManifest != "" {
+		manifestPath, err := ioutil.ReadFile(*baseManifest)
+		if err != nil {
+			log.Fatalf("Failed to read the parent image's manifest: %v", err)
+		}
+		manifestFile := v1.Manifest{}
+		if err = json.Unmarshal([]byte(manifestPath), &manifestFile); err != nil {
+			log.Fatalf("Failed to successfully read manifest file contents: %v", err)
+		}
+	}
+
+	var createTime string
+	if *creationTime == "" {
+		creationTime = nil
+	} else {
+		var unixTime int64
+		if createTime, err = stamp(*creationTime); err != nil {
+			log.Fatalf("Unable to format creation time from BUILD_TIMESTAMP macros: %v", err)
+		}
+		// if creationTime is parsable as a floating point type, assume unix epoch timestamp.
+		// otherwise, assume RFC 3339 date/time format.
+		unixTime, err := strconv.ParseInt(createTime, 10, 64)
+		if err != nil {
+			log.Fatalf("Unable to parse a floating point type from flag creationTime: %v", err)
+		} else {
+			if unixTime > 1.0e+11 {
+				unixTime = unixTime / 1000.0
+			}
+			// construct a RFC 3339 date/time from Unix epoch.
+			t := time.Unix(unixTime, 0)
+			creationTime := t.Format(time.RFC3339)
+			rfcCreationTime, err := time.Parse(time.RFC3339, creationTime)
+			if err != nil {
+				log.Fatalf("Unabel to convert parsed RFC3339 time to time.Time: %v", err)
+			}
+			configFile.Created = v1.Time{rfcCreationTime}
+		}
+	}
+
+	configFile.Author = "Bazel"
+	configFile.OS = *operatingSystem
+	configFile.Architecture = processorArchitecture
+
+	// output['config'] = defaults.get('config', {}) line 161
+	configFile.Config = v1.Config{}
+	if len(entrypoint) > 0 {
+		configFile.Config.Entrypoint = entrypoint
+	}
+	if len(command) > 0 {
+		configFile.Config.Cmd = command
+	}
+	if *user != "" {
+		configFile.Config.User = *user
+	}
+
+	if len(env) != 0 {
+		environMap := keyValueToMap(env)
+		var resolvedValue string
+		for k, v := range environMap {
+			if resolvedValue, err = resolveVariables(v, environMap); err != nil {
```

Comment:
Is this invocation correct? I think you might need to pass in the current os environment for a recursive substitution to work correctly. For example, for something like `PATH=$PATH:foo` I think this will just replace `$PATH` with `$PATH:foo` resulting in the value of `PATH` becoming `$PATH:foo:foo`.

Reply:
seems correct when I tested it against the old one. They output the same content for config.